### PR TITLE
Use ScrambleLogic in AI code.

### LINKED ISSUE #5639: PR: follow-up, Show warning dupe matcher
@asvitkine this warning seems legitimate to fix:

game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java on lines 379..385

```
        PredicateBuilder.of(Matches.territoryIsWater().or(Matches.isTerritoryEnemy(player, data)))
            .and(
                Matches.territoryHasUnitsThatMatch(
                    Matches.unitCanScramble()
                        .and(Matches.unitIsEnemyOf(data, player))
```

We would not want to maintain both pieces of code in parallel, looks like the logic can be made into a `Matches` predicate and consolidated without too much trouble.

_Originally posted by @DanVanAtta in https://github.com/triplea-game/triplea/pull/5619/review_comment/create_

### PR DESCRIPTION
**Pull request by asvitkine**
Use ScrambleLogic in AI code. This is a refactoring that's meant to preserve the same functionality as before, while reusing the code from ScrambleLogic rather than duplicating it.

Among other things, fixes #5639.

## Functional Changes

[] New map or map update
[] New Feature
[] Feature update or enhancement
[] Feature Removal
[X] Code Cleanup or refactor
[] Configuration Change
[] Problem fix:  
[] Other:   

## Testing

[X] Manual testing done

All-AI game on Global 2nd e.

### CODE REVIEW

**Comment by asvitkine** [comment]
In the meanwhile, split off the unrelated fix to:
https://github.com/triplea-game/triplea/pull/5647

**Comment by DanVanAtta** [comment]
@asvitkine PMD violation:
> /home/travis/build/triplea-game/triplea/game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java:367:	Avoid unused local variables such as 'from'.

**Comment by asvitkine** [comment]
> @asvitkine PMD violation:
> 
> > /home/travis/build/triplea-game/triplea/game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java:367:	Avoid unused local variables such as 'from'.

Thanks, I saw it too. I'm holding off on working on this more until we get some of the other bugs I'm finding in AI-only games resolved. I'd like to be able to test this on a clean working state and right now it seems we have some bugs to fix before that.

File: game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java [hunk 1]
```diff
@@ -360,103 +359,33 @@ private static void findScrambleOptions(
...
-                Comparator.<Unit>comparingDouble(
-                        o ->
-                            ProBattleUtils.estimateStrength(
-                                to, List.of(o), new ArrayList<>(), false))
-                    .reversed());
-            canScrambleAir = canScrambleAir.subList(0, maxCanScramble);
-          }
-          moveMap.get(to).getMaxScrambleUnits().addAll(canScrambleAir);
+    final var scrambleLogic = new ScrambleLogic(data, player, moveMap.keySet());
+    for (final var territoryToScramblersEntry :
```
Comment (ron-murhammer) [inline game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java]: Would it be better to specify the type here instead of var since its not very obvious?

File: game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java [hunk 2]
```diff
@@ -360,103 +359,33 @@ private static void findScrambleOptions(
...
+        scrambleLogic.getUnitsThatCanScrambleByDestination().entrySet()) {
+      final Territory to = territoryToScramblersEntry.getKey();
+      for (final var fromToScrambers : territoryToScramblersEntry.getValue().entrySet()) {
```
Comment (ron-murhammer) [inline game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java]: Would it be better to specify the type here instead of var since its not very obvious?

File: game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java]: Jumping in, the first var here, I'm not sure if it would help. The type is convoluted, it's mostly noise. We have an assignment in the following lines, which are probably going to be the thing that is most helpful anyways. Replacing the 2nd var IMO perhaps could be beneficial.

Comment (ron-murhammer) [reply game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java]: Fair point and agree the second is probably more useful than the first. Mostly I just want to have some idea over what I'm iterating over as otherwise its pretty hard to follow.

File: game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java (hunk 2 above)
Comment (asvitkine) [reply game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java]: Done and switched iteration to be on values.

File: game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java (hunk 1 above)
Comment (asvitkine) [reply game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java]: Kept var here, but added type on the inner loop.

**Comment by asvitkine** [comment]
@DanVanAtta do you have any additional feedback? From my side, this is ready to merge. I've even validated by running the old code and new code and verifying results matched.
